test/controllers/api/v2/json_web_tokens_controller_test.rb

```diff
@@ -0,0 +1,109 @@
+require 'test_helper'
+
+class Api::V2::JsonWebTokensControllerTest < ActionController::TestCase
+  def setup
+    @user = FactoryBot.create(:user, admin: false)
+    @role = FactoryBot.create(:role)
+    @generate_jwt = FactoryBot.create(:permission, name: 'generate_json_web_tokens')
+    @invalidate_jwt = FactoryBot.create(:permission, name: 'invalidate_json_web_tokens')
+  end
+
+  #
+  # Generate token tests
+  #
+
+  test "admin generate for self" do
+    as_user(:admin) { post :generate, params: { id: users(:admin).id } }
+    assert_response :success
+  end
+
+  test "admin generate for user" do
+    as_user(:admin) { post :generate, params: { id: users(:one).id } }
+    assert_response :success
+  end
+
+  test "non-admin with permissions generate for self" do
+    @role.add_permissions! [@generate_jwt.name]
+    @user.roles = [@role]
```

Comment:
there is a shorthand for setting up users `setup_user(operation, type, search=nil, user = :one)`
so calling `setup_user('generate', 'json_web_tokens`) would do the job of setting up perms for `users(:one)` (comming from test fixtures) and set the `User.current` to this user.